### PR TITLE
unit test for PCLVisualizer with PointXYZRGB

### PR DESCRIPTION
**Pull request by huanglilong**
unit test for https://github.com/PointCloudLibrary/pcl/issues/5545

### CODE REVIEW

**Comment by larshg** [comment]
Looks good. Could you add the cases that also covers the other places, that @mvieth located?
Then we can add the solutions to this as well.

**Comment by larshg** [comment]
Shouldn't we keep this one @mvieth ? 
It at least covered one of the cases.

**Comment by larshg** [comment]
Could you add a comment above the unit test why this one is run and perhaps a link to the original issue? else its 👍

**Comment by huanglilong** [comment]
@larshg  @mvieth this unit test is for https://github.com/PointCloudLibrary/pcl/issues/5545 to repeat crash in `dynamic_cast<vtkPolyDataMapper*>(am_it->second.actor->GetMapper ())`
maybe we need integration test for `pcl::visualization::PCLVisualizer` to cover almost all code to detect this type on-run crash.
so i think this unit test can be closed. @larshg @mvieth how about yours suggestion?

File: test/visualization/test_visualization.cpp [hunk 1]
```diff
@@ -191,6 +191,34 @@ TEST (PCL, PCLVisualizer_getPointCloudRenderingProperties)
   EXPECT_EQ (b, 0.);
 }
 
+TEST(PCL, PCLVisualizer_updateColorHandlerIndex) {
```
Comment (mvieth) [inline test/visualization/test_visualization.cpp:194]:
```suggestion
// This test was added to make sure the dynamic_cast in updateColorHandlerIndex works correctly (see https://github.com/PointCloudLibrary/pcl/issues/5545)
TEST(PCL, PCLVisualizer_updateColorHandlerIndex) {
```

**Comment by huanglilong** [comment]
@mvieth  updated

**Comment by mvieth** [comment]
I would say it is better to have this test than to not have it. It is true that a lot of code in PCL's visualization module is not covered by automatic tests. If you have suggestions for further tests, I would be interested to see them. But I am not sure what you mean with "integration test"?

**Comment by huanglilong** [comment]
"integration test" just like SIL (software in the loop)
run `pcl::visualization::PCLVisualizer` with real dataset. https://github.com/PointCloudLibrary/pcl/blob/master/tools/pcd_viewer.cpp this tools can be used for this.

this can avoid compile and on-run crash diff merge into master branch.

unit test for algorithm.

**Comment by mvieth** [comment]
I think a real dataset would only very rarely uncover more bugs than just using a few randomly generated points.
Using a tool/an executable like pcd_viewer.cpp for testing wouldn't work so well with our googletest framework. Additionally, we could only test whether the program crashes or not. If we write tests within the googletest framework, we can use the `ASSERT` and `EXPECT`.
In my understanding, an integration test would be used _after_ unit-testing the individual components, to make sure that the components also work correctly together. In the context of the visualization module, I am not sure we can say that we are unit-testing "algorithms", but I believe we can already cover a lot by writing tests for `PCLVisualizer`'s functions in the googletest framework.

**Review by mvieth — APPROVED** [review]

**Pull request merged**